Mage/src/main/java/mage/abilities/keyword/BuybackAbility.java

```diff
@@ -191,7 +191,14 @@ public boolean applies(GameEvent event, Ability source, Game game) {
         if (event.getTargetId().equals(source.getSourceId())) {
             ZoneChangeEvent zEvent = (ZoneChangeEvent) event;
             // if spell fizzled, the sourceId is null
-            return zEvent.getFromZone() == Zone.STACK && zEvent.getToZone() == Zone.GRAVEYARD
+            return zEvent.getFromZone() == Zone.STACK
```

Comment:
curious, what's the purpose of checking toZone == Graveyard anyway?

Reply:
Buyback in the CR is explicitly
> "If the [buyback](https://yawgatog.com/resources/magic-rules/#buyback) [cost](https://yawgatog.com/resources/magic-rules/#cost) was paid, put this [spell](https://yawgatog.com/resources/magic-rules/#spell) into its [owner](https://yawgatog.com/resources/magic-rules/#owner)'s [hand](https://yawgatog.com/resources/magic-rules/#hand) [instead](https://yawgatog.com/resources/magic-rules/#instead) of into that [player](https://yawgatog.com/resources/magic-rules/#player)'s [graveyard](https://yawgatog.com/resources/magic-rules/#graveyard) as it [resolves](https://yawgatog.com/resources/magic-rules/#resolve)."

https://yawgatog.com/resources/magic-rules/#R70227
I think there are corner cases with replacement effect where you may (or maybe sometimes can't if another player has replacement effect affecting you?) return the spell to hand.

The Innocuous Insect does not actually work using CR ruling. But since no other buyback spell will resolve to the battlefield, I was alright tweaking that condition to make the card work.